app/models/teacher.rb

```diff
@@ -45,4 +46,27 @@ def send_magic_link(*)
   def send_devise_notification(notification, *args)
     devise_mailer.send(notification, self, *args).deliver_later
   end
+
+  def should_send_reminder_email?
+    all_reference_requests =
+      ReferenceRequest
+        .joins(:work_history)
+        .where(work_histories: { application_form_id: application_form.id })
+        .where(received_at: nil, expired_at: nil)
+    @remindable_reference_requests =
+      all_reference_requests.select do |reference_request|
+        reference_request.should_send_reminder_email?(
+          reference_request.days_until_expired,
+          reminder_emails.count,
+        )
+      end
```

Comment:
Since we're getting everything from the `reference_request` now, we could change the method in the concern to not take any arguments. An example of how to replace the method in `ReferenceRequest`:

```rb
def should_send_reminder_email?
  (days_until_expired <= 28 && reminder_emails.count.zero?) ||
      (days_until_expired <= 14 && reminder_emails.count == 1)
end
```

This would allow us to remove the special case in the `SendReminderEmail` service, as both teachers and requestables would have a version of this method without arguments.